control-plane/pkg/reconciler/source/v2/sourcev2.go

```diff
@@ -119,6 +123,12 @@ func (r Reconciler) reconcileConsumerGroup(ctx context.Context, ks *sources.Kafk
 						NetSpec: &ks.Spec.KafkaAuthSpec.Net,
 					},
 					Delivery: &internalscg.DeliverySpec{
+						DeliverySpec: &eventingduck.DeliverySpec{
+							Retry:         pointer.Int32(10),
+							BackoffPolicy: &backoffPolicy,
+							BackoffDelay:  pointer.String("PT10S"), // 10 seconds
+							Timeout:       pointer.String("PT500S"),
```

Comment:
Is this too long of a `Timeout`?

Reply:
Ah current `KafkaSource` has a timeout 0s. We should change that too?

Reply:
Hey Ansu, I was talking with @matzew  and the idea is to set timeouts to 600s, I'll update the PR accordingly.

I think 0s (is that no timeout ?) is really bad.

Reply:
Why is that 600s good?

We want that most of the people using this timeout shouldn't need to update this value (and KafkaSource has no config to change it) and, also, because Knative services on big and slow clusters, just to start take ~30 seconds.

Reply:
cc @devguyio in case you want to add your opinion.

Reply:
yes, I agree with a loooooonger timeout - thanks, for bringing this up.

and yes, 600 seems to be a good number, and also: yes, no itention to have users change that a lot.

And yes, `0` is really really bad.

Thanks @aavarghese and @pierDipi

Reply:
Done!

Reply:
Network timeout for TCP/IP are in that range so it seems very reasonable to me to have similiar timeout for eventing delivery for Linux for example is 924 seconds: https://pracucci.com/linux-tcp-rto-min-max-and-tcp-retries2.html
It would be good idea to check and set that default for all Knative Eventing components?

Reply:
> It would be good idea to check and set that default for all Knative Eventing components?

+1, can we keep track of this in an issue in knative/eventing?

Reply:
Opened issue: https://github.com/knative/eventing/issues/6278